### PR TITLE
chore: replace the snapshot chain with enzyme-to-json

### LINKED ISSUE #61: Is React-test-renderer useful ?
My guess is that with Jest and Enzyme, we cover most of our needs.

### PR DESCRIPTION
**Pull request by GuillaumeAmat**
Replace `react-test-renderer` and the other Enzyme's `render` test helper with `shallow` based `enzyme-to-json` snaphots.

Travis seems a bit happier now...

Closes #61

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -54,7 +54,10 @@
     ],
     "moduleNameMapper": {
       "\\.(css|less|svg)$": "identity-obj-proxy"
-    }
+    },
+    "snapshotSerializers": [
+      "enzyme-to-json/serializer"
```
Comment (bleucitron) [inline package.json:59]: What does this `enzyme-to-json` package exactly ?

I write the following:
```
import React from 'react';
import 'jest-styled-components';
import { shallow } from 'enzyme';
import { IconPicker } from 'index';

describe('When using snapshots', () => {
  it('Should render with an element children', () => {
    const wrapper = shallow(<IconPicker />);

    expect(wrapper).toMatchSnapshot();
  });
});
```

without any extra package/conf, and it is ok.

File: src/helpers/tests.js [hunk 2]
```diff
@@ -1,49 +1,27 @@
...
     <DefaultTheme>
       <TestedElement {...props}>
         <div />
       </TestedElement>
     </DefaultTheme>
-  ).toJSON();
+  );
 
-  expect(result).toMatchSnapshot();
+  expect(result.childAt(0)).toMatchSnapshot();
```
Comment (bleucitron) [inline src/helpers/tests.js:16]: Can't we snapshot the whole result ?
With `shallow`, isn't it supposed to be the same ?

**Review by bleucitron — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (GuillaumeAmat) [reply package.json:59]: It gives a better output in the snapshot

File: src/helpers/tests.js (hunk 2 above)
Comment (GuillaumeAmat) [reply src/helpers/tests.js:16]: Without that, the `ThemeProvider` is included in the snapshot, with its `theme` prop...

It is huge *and* we don't test `ThemeProvider` after all, just its child.

**Review by bleucitron — APPROVED** [review]

**Pull request merged**